### PR TITLE
feat: Migrate fledge-export to V2

### PR DESCRIPTION
**Pull request by lenny-intel**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-examples/blob/master/.github/Contributing.md.

## What is the current behavior?
fledge-export not migrated to V2

## What is the new behavior?
fledge-export now migrated to V2

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

File: application-services/custom/fledge-export/res/configuration.toml [hunk 1]
```diff
@@ -1,52 +1,70 @@
...
-        Protocol = "tcp"
+# SecretStore is required when Store and Forward is enabled and running with security
+# so Database credentials can be pulled from Vault.
+# Note when running in docker from compose file set the following environment variables:
+#   - SecretStore_Host: edgex-vault
+[SecretStore]
+Type = 'vault'
+Host = 'localhost'
+Port = 8200
+Path = '/v1/secret/edgex/app-fledge-export/'
```
Comment (AlexCuse) [inline application-services/custom/fledge-export/res/configuration.toml:46]: Should this just be app-fledge-export/ ?

Comment (lenny-intel) [reply application-services/custom/fledge-export/res/configuration.toml:46]: Yup, good catch

**Review by jpwhitemn — CHANGES_REQUESTED** [review]
Lenny - this broke something as part of the export.  I am getting a response from Fledge that the export failed because the correlation-id is an unsupported value type.

level=ERROR ts=2021-07-15T00:57:16.908361807Z app=app-fledge-export source=runtime.go:191 error="export failed: Post \"http://192.168.0.10:6683/sensor-reading\": dial tcp 192.168.0.10:6683: i/o timeout" X-Correlation-ID="unsupported value type" msg="Pipeline function #1 resulted in error"

Let me do some more digging this week.

**Comment by lenny-intel** [comment]
> that the export failed because the correlation-id is an unsupported value type.

Interesting. I haven seen that error , but when I tried to trace it the `X-Correlation-ID` looked fine, but I was not looking at what was put into request header on the exports. I will debug a see what actually put in the request header.

**Comment by lenny-intel** [comment]
> that the export failed because the correlation-id is an unsupported value type.

I traced that done to just a bug in the log message, i.e. call log with function ptr, rather than the result of the function to get the correlation-id. 

The actual error is the timeout.

**Review by jpwhitemn — APPROVED** [review]
All works now - I forgot that I had hardcoded the Fledge Endpoint in config.  In a future version, I should look at that and see if it can be just localhost.

**Pull request merged**